articles/140_topic_and_service_name_mapping.md

```diff
@@ -211,23 +194,51 @@ For example, given the name `{private}foo`, if there is a substitution called `{
+DDS partitions are implemented as an array of strings within the `DDS::Publisher` and `DDS::Subscriber` QoS settings.
+The array allows up to 64 items, and each string in the array is an individual partition.
+This means that every string of the array is an independently matchable entity when paired with the topic name and the other QoS settings.
+Put another way, a DDS DataWriter and DataReader will not connect to one another unless at least one of the partition strings in the array match, even if the topic names match.
+Sometimes special characters are used in the partition strings to express regular expressions matching (such as `+`, `*`, `^`), but they are not officially supported by the standard.
+The maximum size of each string in the array may depend on the DDS or RTPS implementation.
+You can read more about partitions in RTI's documentation:
 
-### ROS Specific Name Prefix
+- [PARTITION_QosPolicy](https://community.rti.com/static/documentation/connext-dds/5.2.3/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/PARTITION_QosPolicy.htm)
 
-In order to differentiate ROS topics easily, all DDS topic names created by ROS shall be prefixed with `rX`, where `X` is a single character that indicates to which subsystem of ROS the topic belongs.
-For example, a plain topic called `/foo` would translate to a DDS topic named `rt__foo`, which is the result of concatenating the prefix `rt` for being a ROS topic, with `__` for the leading `/`, and the topic name `foo`.
+The array of strings have no hierarchy and the order does not matter.
+Each entry in the partition array is directly combined with the DDS topic and they are not sequentially combined.
+If a publisher operates on two partition entries, e.g. `foo` and `bar` with a base name of `baz`, this would result in the moral equivalent of two topics: `/foo/baz` and `/bar/baz`
+Therefore we cannot use the array of strings to represent our implied namespace hierarchy and we have to continue to use the forward slash to accomplish this.
+
+#### Hierarchy with DDS Partitions
+
+Since DDS partitions have no hierarchy or order to the array of strings, this proposal would use only one of the strings in the array to hold the entire ROS name's namespace.
+Therefore we preserve the forward slashes (`/`) that exist in the ROS name's namespace already.
+
+### ROS Specific Namespace Prefix
+
+In order to differentiate ROS topics easily, all DDS topics created by ROS shall be automatically prefixed with a namespace like `/rX`, where `X` is a single character that indicates to which subsystem of ROS the topic belongs.
+For example, a plain topic called `/foo` would translate to a DDS partition `/rt/` and a DDS topic `foo`, which is the result of implicitly adding `/rt` to the namespace of a ROS topic which is in the root namespace `/` and has a base name `foo`.
```

Comment:
@ros2/team If some of you could comment on this idea soon I'd appreciate it. I think it should be addressed before merging this, and I'd like to avoid holding this up any longer.